### PR TITLE
CDAP-4277: Id should not serialize toString and hashCode fields

### PR DESCRIPTION
**Pull request by jwang47**
https://issues.cask.co/browse/CDAP-4277
http://builds.cask.co/browse/CDAP-RBT528

### CODE REVIEW

File: cdap-proto/src/main/java/co/cask/cdap/proto/Id.java [hunk 1]
```diff
@@ -67,8 +67,8 @@ private static boolean isValidNamespaceId(String name) {
     return namespaceMatcher.matchesAllOf(name);
   }
 
-  private String toString;
-  private int hashCode = -1;
```
Comment (anwar6953) [inline cdap-proto/src/main/java/co/cask/cdap/proto/Id.java]: the -1 is depended on by this:
https://github.com/caskdata/cdap/blob/feature/4277/cdap-proto/src/main/java/co/cask/cdap/proto/Id.java#L112

Comment (jwang47) [inline cdap-proto/src/main/java/co/cask/cdap/proto/Id.java]: Fixed.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java [hunk 2]
```diff
@@ -57,6 +57,7 @@
 import co.cask.tephra.TransactionExecutor;
 import co.cask.tephra.TransactionExecutorFactory;
 import com.google.common.base.Charsets;
+import com.google.common.base.Joiner;
```
Comment (anwar6953) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: unused?

**Comment by chtyim** [comment]
LGTM

**Pull request merged**